src/app/features/location/address-search/address-search.component.ts

```diff
@@ -125,12 +128,23 @@ export class AddressSearchComponent implements OnInit {
     );
   }
 
-  selectLocation(selected: GeoResult | string | undefined) {
+  async selectLocation(selected: GeoResult | string | undefined) {
     let result: GeoLocation;
     if (typeof selected === "object") {
       result = { geoLookup: selected };
     } else if (typeof selected === "string") {
-      // special case to set address text from search without mapped location (when no result was found)
+      // Show confirmation dialog instead of custom dialog
+      await this.confirmationDialog.getConfirmation(
+        $localize`No mapped location`,
+        $localize`There is no mapped location for the entered address. You can still save the address as free text.`,
+        [
+          {
+            text: $localize`OK`,
+            dialogResult: true,
+            click: () => {},
+          },
+        ],
```

Comment:
you can simply import `OkButton` from confirmation-dialog.component.ts
```suggestion
        OkButton,
```